src/mlpack/tests/main_tests/linear_svm_test.cpp

```diff
@@ -0,0 +1,866 @@
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(LinearSVMMainTest,
+                         LinearSVMTestFixture);
+
+/**
+ * Ensure that trainingSet are necessarily passed when training.
+ **/
+BOOST_AUTO_TEST_CASE(LinearSVMNoTrainingData)
+{
+  arma::Row<size_t> trainLabels;
+  if (!data::Load("iris_labels.txt", trainLabels))
+    BOOST_FAIL("Cannot load test dataset iris_labels.txt!");
+
+  SetInputParam("labels", std::move(trainLabels));
+
+  // Training data is not provided. Should throw a runtime error.
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Checking that that size and dimensionality of prediction is correct.
+ */
+BOOST_AUTO_TEST_CASE(LinearSVMOutputDimensionTest)
+{
+  arma::mat trainData;
+  if (!data::Load("iris.csv", trainData))
+    BOOST_FAIL("Cannot load test dataset iris.csv!");
+
+  arma::Row<size_t> trainLabels;
+  if (!data::Load("iris_labels.txt", trainLabels))
+    BOOST_FAIL("Cannot load test dataset iris_labels.txt!");
+
+  arma::mat testData;
+  if (!data::Load("iris_test.csv", testData))
+    BOOST_FAIL("Cannot load test dataset iris_test.csv!");
+
+  size_t testSize = testData.n_cols;
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(trainLabels));
+  SetInputParam("test", std::move(testData));
+
+  // Training the model.
+  mlpackMain();
+
+  // Get the output predictions of the test data.
+  const arma::Row<size_t>& testLabels =
+      CLI::GetParam<arma::Row<size_t>>("predictions");
+
+  // Output predictions size must match the test data set size.
+  BOOST_REQUIRE_EQUAL(testLabels.n_rows, 1);
+  BOOST_REQUIRE_EQUAL(testLabels.n_cols, testSize);
+}
+
+/**
+  * Ensuring that the labels size is checked.
+ **/
+BOOST_AUTO_TEST_CASE(LinearSVMCheckLabelsSizeTest)
+{
+  arma::mat trainData;
+  if (!data::Load("iris.csv", trainData))
+    BOOST_FAIL("Cannot load test dataset iris.csv!");
+
+  arma::Row<size_t> trainLabels;
+  if (!data::Load("vc2_labels.txt", trainLabels))
+    BOOST_FAIL("Cannot load test dataset vc2_labels.txt!");
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(trainLabels));
+
+  // Labels with incorrect size. It should throw a runtime error.
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Checking two options of specifying labels (extra row in train matrix and
+ * extra parameter) and ensuring that predictions are the same.
+ */
+BOOST_AUTO_TEST_CASE(LinearSVMLabelsRepresentationTest)
+{
+  arma::mat trainData1({{1.0, 2.0, 3.0}, {1.0, 4.0, 9.0}, {0, 1, 1}});
+  arma::mat testData({{4.0, 5.0}, {1.0, 6.0}});
+
+  SetInputParam("training", std::move(trainData1));
+  SetInputParam("test", testData);
+
+  // The first solution.
+  mlpackMain();
```

Comment:
Since the initial point for the optimization is random, should we set a fixed random seed for both calls here?

Reply:
Added